src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java

```diff
@@ -267,13 +269,27 @@ public SkyValue compute(SkyKey key, Environment env) throws ConfiguredTargetFunc
     //  would exit this SkyFunction and restart it when permits were available.
     acquireWithLogging(key);
     try {
+      // Determine what toolchains are needed by this target.
+      unloadedToolchainContexts =
+          computeUnloadedToolchainContexts(
+              env,
+              ruleClassProvider,
+              defaultBuildOptions,
+              ctgValue,
+              configuredTargetKey.getToolchainContextKey());
+      if (env.valuesMissing()) {
+        return null;
+      }
+
       // Get the configuration targets that trigger this rule's configurable attributes.
       ImmutableMap<Label, ConfigMatchingProvider> configConditions =
           getConfigConditions(
-              ctgValue.getTarget(),
               env,
               ctgValue,
               transitivePackagesForPackageRootResolution,
+              unloadedToolchainContexts == null
+                  ? null
+                  : unloadedToolchainContexts.getDefaultToolchainContext().targetPlatform(),
```

Comment:
All toolchain contexts for the same configured target, regardless of exec group, should have the same target platform (right @juliexxia ?)

We should consider enshrining that in the API by adding a unloadedToolchainContexts.targetPlatform() method.

Reply:
Yeah that sounds right and looks right according to https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java;l=531?q=ConfiguredTargetFunction&ss=bazel

@katre I like that idea. Do you think there's anything else to discuss there? If not I can mail something.

Reply:
I don't think so, it makes sense. Go ahead if you have time, I was going to file an issue and then deal with it later :)

Reply:
To be clear, this suggestion is for a followup PR?

Reply:
Yes, nothing to do in this PR.